frontend/public/components/monitoring/utils.ts

```diff
@@ -144,3 +144,12 @@ export const targetSource = (target: Target): AlertSource =>
   target.labels?.prometheus === 'openshift-monitoring/k8s'
     ? AlertSource.Platform
     : AlertSource.User;
+
+const TWO_DAYS_MS = 172800000;
+
+export const timeoutFromTimespan = (timespan: number): number => {
+  if (timespan >= TWO_DAYS_MS) {
+    return 120 * 1000;
+  }
+  return 60 * 1000;
+};
```

Comment:
```suggestion
export const timeoutFromTimespan = (timespan: number): number =>
  (timespan >= TWO_DAYS_MS ? 120 : 60) * 1000;
```
Nit

Reply:
More fundamentally, I'm not sure that basing the timeout on the timespan is the right approach. We request the same number of data samples regardless of the timespan, so the amount of data returned is about the same for a 1 hour span or a 1 week span. So we don't expect a large difference in the response time.

Also, 120 seconds seems like a lot to me.

I'm wondering if we could radically simplify this whole change by just increasing the default timeout. E.g. always use `60s` instead of the current `30s`?

Reply:
The samples and data we get is the same, but the volume of data aggregated by thanos/prometheus is greater with a longer timespan, hence the request takes longer. From the bug description:

```
Call to /api/prometheus/api/v1/query_range?
start=.......&end=.....&step=40320&query=historgram_quantile%280.99%2C+sum%28rate%28etcd_request_druration_seconds_bucket%5B5m%5D%29%29+by%28operation%2Cle%29%29&timeout=30s timed out after 60000ms
```

the request was failing already with the default 60s timeout we have for the console fetch, and the timeout set on the query was 30s. So this fix aimed for making the timeouts consistent and increasing it on a longer timespan. I also thought about extending the timeout default for dashboards requests, but that change will be similar to this as the timeout on the console fetch was not passed through using the hook. I took the 120s timeout based on Arun's [comment](https://bugzilla.redhat.com/show_bug.cgi?id=2070596#c13) on the initial bug. 

The more radical approach will be to increase the console timeout to 120s, but this will apply to all requests.

Reply:
I think that using `60s` for all dashboard requests is a good compromise.

Can we just define a `DASHBOARDS_QUERY_TIMEOUT` constant and use that when we call `getPrometheusURL` instead of passing `timeout` around?

Reply:
The need to pass `timeout` around was due to the fact that the consoleFetch has already a 60s timeout by default and didn't allow to pass a different value. But now that we are setting the same default ( 60s and not 120s) then we don't need to pass it around.